// ember-cli-cordova/tmp/class-tmp_cache_dir-MzMqWqDi.tmp/dummy/routes/nav-bar/options-from-model.js
define("dummy/routes/nav-bar/options-from-model", 
  ["ember","ember-cli-cordova/mixins/routes/nav-bar","exports"],
  function(__dependency1__, __dependency2__, __exports__) {
    "use strict";
    var Ember = __dependency1__["default"];
    var NavBarMixin = __dependency2__["default"];

    __exports__["default"] = Ember.Route.extend(NavBarMixin, {
      nav: {
        controller: 'nav-bar',
        title: {
          text: function(model) {
            return model.get('title');
          }
        },

        leftButton: {
          text: function(model) {
            return model.get('leftButton');
          },
          icon: function(model) {
            return model.get('leftClass');
          }
        },

        rightButton: {
          text: function(model) {
            return model.get('rightButton');
          },
          icon: function(model) {
            return model.get('rightClass');
          }
        }
      },

      model: function() {
        return Ember.Object.create({
          title:        'modelOption',
          leftButton:   'modelLeft',
          leftClass:    'leftClass',
          rightButton:  'modelRight',
          rightClass:   'rightClass'
        });
      }
    });
  });
